src/js/components/OptionalPortsAndServiceDiscoveryComponent.jsx

```diff
@@ -32,9 +32,14 @@ function determinePortDefinitionsType(fields) {
 
 function isTooComplexStructure(fields) {
   return fields.portDefinitions != null &&
-      fields.portDefinitions.some(portDefinition =>
-        portDefinition.labels != null &&
-      !!Object.keys(portDefinition.labels).length
+      fields.portDefinitions.some(portDefinition => {
+        var definition = Object.assign({}, portDefinition);
+        if (definition.labels != null) {
+          delete definition.labels["VIP_0"];
```

Comment:
Indeed the cloning happens on the wrong object. Thanx for pushing me to a second look at my own functions. :)